subsys/mpsl/fem/common/mpsl_fem_utils.c

```diff
@@ -27,7 +27,7 @@ int mpsl_fem_utils_ppi_channel_alloc(uint8_t *ppi_channels, size_t size)
 
 	for (int i = 0; i < size; i++) {
 		IF_ENABLED(CONFIG_HAS_HW_NRF_PPI,
-			(err = nrfx_ppi_channel_alloc(&ppi_channels[i]);));
+			(err = nrfx_ppi_channel_alloc((nrf_ppi_channel_t *)&ppi_channels[i]);));
```

Comment:
Have a look on prototype:
```c
nrfx_err_t nrfx_ppi_channel_alloc(nrf_ppi_channel_t * p_channel);
```
The parameter is pointer to `nrf_ppi_channel_t`. The `nrf_ppi_channel_t` is an enum type.
It is implementation-defined what is an underlying type of an `enum` type
Please look [here](https://en.cppreference.com/w/c/language/enum.html#:~:text=It%20is%20implementation%2Ddefined%20which%20type%20is%20compatible%20with%20any%20given%20enumerated%20type%2C%20but%20whatever%20it%20is%2C%20it%20must%20be%20capable%20of%20representing%20all%20enumerator%20values%20of%20that%20enumeration.)
. 
Possibly if you compile with clang it is in fact `int` not the `uint8_t`, casting to get rid of warning is a bad idea here, unless you are positive that underlying type of `nrf_ppi_channel_t` the same as the underlying type of `uint8_t`. 

What I would recommend here is
```suggestion
#if (CONFIG_HAS_HW_NRF_PPI)
			nrf_ppi_channel_t ch;
			err = nrfx_ppi_channel_alloc(ch);
			ppi_channels[i] = (uint8_t)ch;
#endif
```